jsettlers.graphics/src/jsettlers/graphics/map/controls/original/panel/content/Label.java

```diff
@@ -20,12 +20,18 @@
 import jsettlers.graphics.utils.UIPanel;
 
 public class Label extends UIPanel {
+    public enum HorizontalAlignment {
+        LEFT,
+        CENTRE,
+        RIGHT,
+    }
 
 	private final EFontSize size;
-	private final String[] words;
+	private String[] words;
```

Comment:
We can't do this for this game. There are so many changes to the game grid by logic that we cannot afford to pass them all through a queue system. And we cannot afford to pause logic every time the graphic gets drawn. This is why we accept that the UI is not completely synchronized (as it would be in a well-written Swing application) but instead we accept that for a small fraction of a second, an "invalid" state in the middle of an update might be shown. We often have the case that movables at the top of the screen are one tick behind the ones at the bottom. The user won't see the difference. But we need to ensure that no crashes (Array out of bounds, ...) result from this.